### PR TITLE
Port tests over to RSpec

### PR DESCRIPTION
**Pull request by ashfire908**
Replaces old test suite. Optimization has been done to minimize setup/teardown of test schema to only what is needed. Should greatly improve test speed and reliability.

Included changes beyond the port:

* Fix error data parsing failure handling
* Fully offload json serialization to `FaunaJson`
* Minor cleanup

### CODE REVIEW

File: spec/query_spec.rb [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+    it 'recursively wraps hashes' do
+      expr = Fauna::Query.expr { { x: 1, y: { foo: 2 }, z: add(1, 2) } }
+      query = { object: { x: 1, y: { object: { foo: 2 } }, z: { add: [1, 2] } } }
+
+      expect(to_json(expr)).to eq(to_json(query))
+    end
+
+    it 'recursively wraps special types' do
+      expr = Fauna::Query.expr { { x: { y: Time.at(0).utc } } }
+      query = { object: { x: { object: { y: { :@ts => Time.at(0).utc.iso8601(9) } } } } }
```
Comment (freels) [inline spec/query_spec.rb]: heh. Could you hardcode the timestamp string here since it isn't changing?

Comment (ashfire908) [inline spec/query_spec.rb]: Sure.

**Comment by freels** [comment]
shipit modulo note above.

**Comment by freels** [comment]
:sheep:

**Pull request merged**